src/plugin.ts

```diff
@@ -32,9 +32,56 @@ export async function startStopTask(inputs: PluginInputs, env: Env) {
       } else {
         errorMessage = context.logger.error(`Failed to run comment evaluation. ${err}`, { err });
       }
-      await addCommentToIssue(context, `${errorMessage?.logMessage.diff}\n<!--\n${JSON.stringify(errorMessage?.metadata, null, 2)}\n-->`);
+
+      await addCommentToIssue(context, `${sanitizeDiff(errorMessage?.logMessage.diff)}\n<!--\n${sanitizeMetadata(errorMessage?.metadata)}\n-->`);
     }
   } else {
     context.logger.error(`Unsupported event: ${context.eventName}`);
   }
 }
+
+function sanitizeDiff(diff?: LogReturn["logMessage"]["diff"]): string {
+  if (!diff) return "";
+  // eslint-disable-next-line no-useless-escape
+  const backticks = diff.match(/\`\`\`/g);
+  if (!backticks) return "```\n" + diff + "\n```";
```

Comment:
why do we need to add ```, the diff already has that?

Reply:
I know I thought this myself but when gentlementlegen mentioned that the comment would break I realised it would also break if somehow there was only set of backticks although I do agree it's pretty pointless.

Agree to remove?

Reply:
@gentlementlegen rfc

Reply:
I want to add in case it's not clear, I mean to remove me sanitizing the diff as it's very likely not needed but the metadata object sanitizing should remain

Reply:
If the backticks are not escaped they indeed could probably break the comment. It would be safer to transform them to their related code `&#96` if you intend to include them in GitHub comments, which would be safer. Otherwise our logger should take care of adding these around the comment body.

Reply:
escaping them breaks formatting when posted, edit this message to see. I think we could just remove it because the logger is guaranteed to diff it correctly. The only way I could break formatting is if a set of backticks between between two new lines exists which I don't think is possible from anything that we log in the initial string. So I'll remove it.

&#96;&#96;&#96;diff
! Failed to run comment evaluation.
&#96;&#96;&#96;
<!--
{
  &quot;err&quot;: &quot;Error&quot;,
  &quot;caller&quot;: &quot;startStopTask&quot;
}
-->